.github/workflows/run-ui-tests.yml

```diff
@@ -44,22 +44,100 @@ jobs:
 
       # Setup Gradle
       - name: Setup Gradle
-        uses: gradle/actions/setup-gradle@v4
+        uses: gradle/actions/setup-gradle@v4.2.2
         with:
           cache-read-only: true
 
-      # Run IDEA prepared for UI testing
-      - name: Run IDE
-        run: ${{ matrix.runIde }}
+      # Set up Xvfb for Linux headless mode
+      - name: Setup Xvfb for Linux
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt-get update -y
+          sudo apt-get install -y xvfb x11-utils libxkbcommon-x11-0 libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0 libxcb-xinerama0 libxcb-xfixes0
+          export DISPLAY=:99.0
+          Xvfb -ac :99 -screen 0 1920x1080x16 &
+          sleep 3
+          echo "DISPLAY=:99.0" >> $GITHUB_ENV
+
+      # Run UI test with IDE Starter framework
+      - name: Run UI Test
+        env:
+          CODECOV_API_TOKEN: ${{ secrets.CODECOV_API_TOKEN }}
+          LICENSE_KEY: ${{ secrets.LICENSE_KEY }}
+          UI_BRANCH_NAME: "${{ matrix.osName }}-${{ github.ref_name }}"
+          DISPLAY: ${{ runner.os == 'Linux' && ':99.0' || '' }}
+        run: |
+          echo "UI on ${{ matrix.osName }} run #: ${{ github.run_number }} and id: ${{ github.run_id }}"
+          ./gradlew uiTest  
+        timeout-minutes: 30
+
+      # List build directory for debugging
+      - name: List build directory structure
+        if: always()
+        shell: bash
+        run: |
+          echo "=== Build directory structure ==="
+          if [ -d "build" ]; then
+            find build -type d -name "log" -o -name "screenshots" -o -name "ui-hierarchy" 2>/dev/null | head -20 || true
+            echo "=== Looking for IDE test output ==="
+            find build -type d -path "*/ide-tests/*" 2>/dev/null | head -20 || true
+          else
+            echo "Build directory not found"
+          fi
+
+      - name: Collect Test Results
```

Comment:
This can be removed if we update SetUp.kt to use full CI integration with the DI framework.